### ISSUE TITLE
ExecuteContext.batchRows should match BatchUpdateException's getUpdateCounts

### ISSUE THREAD

**Issue by xathien** [jOOQ/jOOQ#5265]
When executing queries in batch, if one query fails, it can be useful to figure out which one it was. jOOQ's ExecuteContext seems like the right place to do this, but the native function `batchRows` returns -1 for all queries in the batch, not just the failed ones. The nested BatchUpdateException still has `getUpdateCounts`, fortunately, and that can be used to find the actual failure, as follows:

```java
public class JooqExceptionTranslator extends DefaultExecuteListener
{
    @Override
    public void exception(ExecuteContext ctx)
	{
		SQLException sqlException = ctx.sqlException();
		if (sqlException == null) // We were called for a RuntimeException instead; just disregard
			return;

        SQLExceptionTranslator translator = new SQLErrorCodeSQLExceptionTranslator(MYSQL.name());
		if (sqlException instanceof BatchUpdateException)
		{
			BatchUpdateException buex = (BatchUpdateException)sqlException;
			String[] queries = ctx.batchSQL();
			StringJoiner failedSQL = new StringJoiner(" | ");
			int[] updateCounts = buex.getUpdateCounts();
			for (int i = 0; i < updateCounts.length; i++)
			{
				if (updateCounts[i] == Statement.EXECUTE_FAILED)
					failedSQL.add(queries[i]);
			}
			ctx.exception(translator.translate("jOOQ", failedSQL.toString(), sqlException));
		}
		else
			ctx.exception(translator.translate("jOOQ", join(ctx.batchSQL(), " | "), sqlException));
    }
}
```

This is a bit unwieldy, and it seems like jOOQ could make `ExecuteContext::batchRows` follow the conventions of `BatchUpdateException::getUpdateCounts`, or maybe make it more directly available through another similar function. Maybe it could be made even more transparent by providing an `ExecuteContext::<String[]>batchFailedQueries()`?

Thanks again for a great product, Lukas!

**Comment by lukaseder** [comment]
Thank you very much for the suggestion. Indeed, we should expose this information in an easier way. I'll think of something...

**Comment by lukaseder** [comment]
... hah, we already have a `batchRows()` method, so this is a bug, then.

**Comment by knutwannheden** [comment]
Yes, `BatchSingle#executePrepared()` for instance, should have a dedicated exception handler for `BatchUpdateException` which would extract the update counts and make them available through the `ExecuteContext#batchRows()`.

AFAIK it is up to the driver / database if any rows will be processed after the first error or not.
